datajoint/fetch.py

```diff
@@ -33,36 +46,59 @@ class Fetch:
 
-    def __call__(self, *attrs, offset=None, limit=None, order_by=None, as_dict=False, squeeze=False):
+    def __call__(self, *attrs, offset=None, limit=None, order_by=None, format=None, as_dict=False, squeeze=False):
         """
         Fetches the expression results from the database into an np.array or list of dictionaries and unpacks blob attributes.
 
         :param attrs: zero or more attributes to fetch. If not provided, the call will return
         all attributes of this relation. If provided, returns tuples with an entry for each attribute.
         :param offset: the number of tuples to skip in the returned result
         :param limit: the maximum number of tuples to return
-        :param order_by: the list of attributes to order the results. No ordering should be assumed if order_by=None.
+        :param order_by: a single attribute or the list of attributes to order the results.
+                No ordering should be assumed if order_by=None.
+                To reverse the order, add DESC to the attribute name or names: e.g. ("age DESC", "frequency")
+                To order by primary key, use "KEY" or "KEY DESC"
+        :param format: Effective when as_dict=False and when attrs is empty
+                None: default from config['fetch_format'] or 'array' if not configured
+                "array": use numpy.key_array
+                "frame": output pandas.DataFrame. .
         :param as_dict: returns a list of dictionaries instead of a record array
         :param squeeze:  if True, remove extra dimensions from arrays
         :return: the contents of the relation in the form of a structured numpy.array or a dict list
         """
 
-        # if 'order_by' passed in a string, make into list
-        if isinstance(order_by, str):
-            order_by = [order_by]
+        if order_by is not None:
+            # if 'order_by' passed in a string, make into list
+            if isinstance(order_by, str):
+                order_by = [order_by]
+            # expand "KEY" or "KEY DESC"
+            order_by = list(_flatten_attribute_list(self._expression, order_by))
 
         # if attrs are specified then as_dict cannot be true
         if attrs and as_dict:
             raise DataJointError('Cannot specify attributes to return when as_dict=True. '
-                                 'Use proj() to select attributes or set as_dict=False')
+                                 'Use '
+                                 'proj() to select attributes or set as_dict=False')
+        # format should not be specified with attrs or is_dict=True
+        if format is not None and (as_dict or attrs):
+            raise DataJointError('Cannot specify output format when as_dict=True or '
+                                 'when attributes are selected to be fetched separately.')
+
+        if format not in {None, "array", "frame"}:
+            raise DataJointError('Fetch output format must be in {"array", "frame"}')
```

Comment:
I always appreciate error message that tells me the wrong value I gave. So could we modify this to say 

```suggestion
            raise DataJointError('Fetch output format must be in {{"array", "frame"}} but "{}" was given'.format(format))
```

Reply:
agreed.